python/hsfs/training_dataset.py

```diff
@@ -286,11 +286,11 @@ def show(self, n: int, split: str = None):
         """
         self.read(split).show(n)
 
-    def add_tag(self, name: str, value: str = None):
+    def add_tag(self, name: str, value: str):
```

Comment:
Same comment for feature groups.

Reply:
removed the `str` constrain, allowing any object that can have `json.dumps` applied to it correctly
adding a ruby test in hopsworks for this...